### PR TITLE
[runtime] Call unhandled exception handler explicitly to fix for bitcode

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -745,7 +745,14 @@ static guint32 WINAPI start_wrapper_internal(void *data)
 		args [0] = start_arg;
 		/* we may want to handle the exception here. See comment below on unhandled exceptions */
 		mono_runtime_delegate_invoke_checked (start_delegate, args, &error);
-		mono_error_raise_exception (&error); /* OK, triggers unhandled exn handler */
+
+		if (!mono_error_ok (&error)) {
+			MonoException *ex = mono_error_convert_to_exception (&error);
+			mono_error_cleanup (&error);
```
Comment (vargaz) [inline mono/metadata/threads.c]: This is already done by convert_to_exception ().

Comment (vargaz) [inline mono/metadata/threads.c]: You can also do:
ex = convert_to_exception ()
if (ex)
   mono_unhandled_exception (ex);

**Comment by alexanderkyte** [comment]
I've pushed the requested changes

**Comment by vargaz** [comment]
Where are those ?

**Comment by alexanderkyte** [comment]
The convert_to_exception usage change in the previous comments

**Comment by vargaz** [comment]
Ah, ok. LGTM.

**Pull request merged**